packages/wallet/src/redux/protocols/transaction-submission/actions.ts

```diff
@@ -1,52 +1,60 @@
+import { WalletProtocol } from '../../types';
+import { BaseProcessAction } from '../actions';
+
 export type TransactionAction =
   | TransactionSentToMetamask
   | TransactionSubmissionFailed
   | TransactionSubmitted
   | TransactionConfirmed
   | RetryTransaction;
 
-export const TRANSACTION_SENT_TO_METAMASK = 'WALLET.COMMON.TRANSACTION_SENT_TO_METAMASK';
-export const TRANSACTION_SUBMISSION_FAILED = 'WALLET.COMMON.TRANSACTION_SUBMISSION_FAILED';
-export const TRANSACTION_SUBMITTED = 'WALLET.COMMON.TRANSACTION_SUBMITTED';
-export const TRANSACTION_CONFIRMED = 'WALLET.COMMON.TRANSACTION_CONFIRMED';
-export const RETRY_TRANSACTION = 'WALLET.COMMON.RETRY_TRANSACTION';
+export const TRANSACTION_SENT_TO_METAMASK = 'WALLET.TRANSACTION_SENT_TO_METAMASK';
+export const TRANSACTION_SUBMISSION_FAILED = 'WALLET.TRANSACTION_SUBMISSION_FAILED';
+export const TRANSACTION_SUBMITTED = 'WALLET.TRANSACTION_SUBMITTED';
+export const TRANSACTION_CONFIRMED = 'WALLET.TRANSACTION_CONFIRMED';
+export const RETRY_TRANSACTION = 'WALLET.RETRY_TRANSACTION';
+export const TRANSACTION_FINALIZED = 'WALLET.TRANSACTION_FINALIZED';
 
-export interface TransactionSentToMetamask {
+export interface TransactionSentToMetamask extends BaseProcessAction {
   type: typeof TRANSACTION_SENT_TO_METAMASK;
+  protocol: WalletProtocol.TransactionSubmission;
   channelId: string;
   processId: string;
 }
-export interface TransactionSubmissionFailed {
+export interface TransactionSubmissionFailed extends BaseProcessAction {
   type: typeof TRANSACTION_SUBMISSION_FAILED;
+  protocol: WalletProtocol.TransactionSubmission;
   channelId: string;
   processId: string;
   error: { message?: string; code };
 }
 
-export interface TransactionSubmitted {
+export interface TransactionSubmitted extends BaseProcessAction {
   type: typeof TRANSACTION_SUBMITTED;
+  protocol: WalletProtocol.TransactionSubmission;
   channelId: string;
   processId: string;
   transactionHash: string;
 }
 
-export interface TransactionConfirmed {
+export interface TransactionConfirmed extends BaseProcessAction {
   type: typeof TRANSACTION_CONFIRMED;
+  protocol: WalletProtocol.TransactionSubmission;
```

Comment:
Can we use a shared common interface for the shared properties? Or have we decided we want to redeclare the properties on each interface?

Reply:
This is cherry-picked from Tom's branch -- the protocol should actually be removed from these actions, since they have a process id. (The protocol lives on the process state)